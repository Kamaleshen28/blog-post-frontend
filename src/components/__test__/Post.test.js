import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import Post from "../Post";

const mockData = {
    "date": "2nd Januray, 2018",
    "readingTime": "2 mins",
    "title": "The future of abstract art and the culture ...",
    "description": "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    "claps": 10,
    "liked": false,  
    "image": "./abstract.png"
}
const handleClickClap = jest.fn();
const handleClickHeart = jest.fn();
// Just check whether the function is called once or not

describe('Post', () => {

    it('should render the post correctly', async() => {
        const {asFragment} = render(<Post data={mockData} handleClickClap={handleClickClap} handleClickHeart={handleClickHeart}/>)
        expect(screen.getByText('2nd Januray, 2018')).toBeTruthy();
        expect(asFragment()).toMatchSnapshot();
    })
    it('should call handleClickHeart when heart is clicked', ()=>{
        render(<Post data={mockData} handleClickClap={handleClickClap} handleClickHeart={handleClickHeart} />);
        const likeButton = screen.queryByTestId('heart-toggle');
        fireEvent.click(likeButton);
        expect(handleClickHeart).toHaveBeenCalled()
    })
    it('should call handleClickClap when the clap icon is clicked', ()=>{
        render(<Post data={mockData} handleClickClap={handleClickClap} handleClickHeart={handleClickHeart} />);
        const likeButton = screen.queryByTestId('clap-btn');
        fireEvent.click(likeButton);
        expect(handleClickClap).toHaveBeenCalled()
    })
})