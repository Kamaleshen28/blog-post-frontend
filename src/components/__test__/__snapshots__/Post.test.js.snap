// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render the post correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div>
      <img
        class="card--img"
        src="./abstract.png"
      />
    </div>
    <div>
      <div
        class="date"
      >
        <a>
          2nd Januray, 2018
        </a>
        <a>
          2 mins
        </a>
      </div>
      <div>
        <h3
          class="title"
        >
          The future of abstract art and the culture ...
        </h3>
      </div>
      <div>
        <a
          class="desc"
        >
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </a>
      </div>
      <div>
        <hr
          class="line"
        />
      </div>
      <div
        class="last--card--section"
      >
        <div
          class="clap-section"
        >
          <img
            alt="clap"
            class="clap--img"
            data-testid="clap-btn"
            src="./clapping.svg"
          />
          <a
            data-testid="clap-count"
          >
            10
          </a>
        </div>
        <div
          class="heart-section"
        >
          <img
            alt="heart"
            class="heart--img"
            data-testid="heart-toggle"
            src="./heart-black.svg"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
